ui-tests/src/test/java/io/syndesis/qe/pages/integrations/edit/IntegrationFlowViewComponent.java

```diff
@@ -99,15 +102,39 @@ public void clickRandomTrash() {
         this.getElementRandom(Element.DELETE).shouldBe(visible).click();
     }
 
-    public void clickAddStepLink(int pos){
+    public void clickAddStepLink(int pos) {
 
-        List<WebElement> allStepInserts = getRootElement().findElements(Element.STEP_INSERT);
-        WebElement stepElement = allStepInserts.get(pos);
-        Actions action = new Actions(WebDriverRunner.getWebDriver());
-        //to make ADD_STEP element visible:
-        action.moveToElement(stepElement);
+        List<SelenideElement> allStepInserts = getRootElement().$$(Element.STEP_INSERT)
+                .shouldHave(CollectionCondition.sizeGreaterThanOrEqual(pos));
+        SelenideElement stepElement = allStepInserts.get(pos);
+
+        stepElement.hover();
 
         getRootElement().$(Link.ADD_STEP).shouldBe(visible).click();
     }
 
+    /**
+     * If you only have 3 steps - start, finish and a step in the middle to get the middle step
+     * set stepPosition to "middle". Otherwise set it to start or finish to get first/last step.
+     *
+     * @param stepPosition
+     * @return text in popover element
+     */
+    public String checkTextInHoverTable(String stepPosition) {
+        String text;
+        if (stepPosition.equalsIgnoreCase("middle")) {
+            // This is ugly but for now it works - it has only one usage: for getting our data-mapper step
+            // which is between start and finish step.
+            // Explanation: we have 3 steps and between them 2 elements with insert step option --> 5 total
+            // with class "step" and we want the third element
+            $$(By.className("step")).shouldHaveSize(5).get(2)
```

Comment:
I been wondering if those 2 numbers, `5` and `2` shouldn't be rather parameters of method?

Reply:
Just to make it a bit less ugly :)